### PR TITLE
pybats.kyoto: Fix histogram-style Kp plots.

### LINKED ISSUE #252: pybats: Kp histogram-style plots produce poor/incorrect results.
When using spacepy.pybats.kyoto.KyotoKp object's add_histplot function, instead of producing a continuous, single-valued, step-function-like plot (as is expected for the 3-hour Kp index), the line crosses over itself, duplicating previous times in an obviously incorrect fashion.

I will push solution shortly.

### Minimal example to reproduce issue:

```
from spacepy.pybats import kyoto
a = kyoto.fetch('kp', (2003, 10), (2003, 10))
a.add_histplot()
```

No traceback, just discontinuous/scattered results.

### OS, Python version, and dependency version information:
Issue is platform independent.

### PR DESCRIPTION
**Pull request by spacecataz**
Fixed an issue where Kp plots from pybats.kyoto where incorrect.  Underlying issue was creation of the time boundaries on bins; this has been rectified.  Added a small change to how legend labeling for these plots is done within spacepy.pybats.BatsLog and GeoIndexFile objects.

## PR Checklist
- [x] New code has inline comments where necessary
- [x] Any new modules, functions or classes have docstrings consistent with SpacePy style
- [x] Added an entry to CHANGELOG if fixing a major bug or providing a major new feature
- [ ] New features and bug fixes should have unit tests

Closes #252

### CODE REVIEW

**Comment by spacecataz** [comment]
Changed approach via @drsteve's suggestions, should be good to go now.

**Comment by drsteve** [comment]
Fixed CHANGELOG conflict...

**Comment by jtniehof** [comment]
@drsteve , now that you have your fingerprints on it :) are you happy and should I do a quick check and merge?

**Comment by drsteve** [comment]
I do have a quick question for @spacecataz ... let me see if I can get an answer offline, might be faster.

**Comment by drsteve** [comment]
Further updates added:
- Fixed end bins so the full extent of the end bins are shown (not just half-bins)
- Added option to use `spacepy.plot.levelPlot` to give a filled Kp step-plot (with kwarg to pass extra arguments)
- Fixed retired kwarg ('on'/'off' in Axes.grid) used in levelPlot that was leading to inconsistent grid behavior
- Updated docstring to reflect current usage and remove deprecated keywords

@spacecataz - can you take a look? Ping me if you have any issues (since I force pushed to your branch!)

File: spacepy/pybats/kyoto.py [hunk 1]
```diff
@@ -234,41 +234,38 @@ def add_histplot(self, target=False, loc=111, label='Kyoto $K_{p}$',
...
+
         Extra keyword arguments are passed to :function:`matplotlib.pyplot.plot`
-        to customize the line style.
+        to customize the line style (and are ignored if filled is True).
 
         Examples
         ========
+        >>> import matplotlib.pyplot as plt
         >>> import spacepy.pybats.kyoto as kt
         >>> kp = kt.fetch('kp', (1981, 11), (1981, 11)
```
Comment (jtniehof) [inline spacepy/pybats/kyoto.py]:
```suggestion
        >>> kp = kt.fetch('kp', (1981, 11), (1981, 11))
```

File: spacepy/pybats/kyoto.py [hunk 2]
```diff
@@ -234,41 +234,38 @@ def add_histplot(self, target=False, loc=111, label='Kyoto $K_{p}$',
...
-
-        fig, ax = set_target(target, figsize=(10,4), loc=loc)
-        line=ax.plot(newtime, newkp, label=label, **kwargs)
+        fig, ax = set_target(target, figsize=(10,4),  loc=loc)
+        if not filled:
+            tvar = self['binstart'].tolist()
+            tvar.append(tvar[-1]+dt.timedelta(hours=3))
+            kpvar = self['kp'].tolist()
+            kpvar.append(kpvar[-1])
+            line = ax.plot(tvar, kpvar, label=label, ds='steps-post', **kwargs)
```
Comment (jtniehof) [inline spacepy/pybats/kyoto.py]:
```suggestion
            line = ax.plot(tvar, kpvar, label=label, drawstyle='steps-post',
            **kwargs)
```

Comment (jtniehof) [reply spacepy/pybats/kyoto.py]: `ds` was added in matplotlib 3.1.0 (and then a linebreak to keep under 80 columns).

**Comment by jtniehof** [comment]
If @spacecataz is happy with @drsteve 's changes and can incorporate my suggestions, happy to merge.

**Comment by drsteve** [comment]
Suggestions look good to me. @spacecataz, can you take a look? If everything looks good then please add changes from @jtniehof's review and approve or merge!

**Pull request merged**